modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/claims/pages/ClaimsTable/index.tsx

```diff
@@ -0,0 +1,261 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import ClayButton from '@clayui/button';
+import ClayIcon from '@clayui/icon';
+import {ClayPaginationWithBasicItems} from '@clayui/pagination';
+import ClayPaginationBar from '@clayui/pagination-bar';
+import {useEffect, useState} from 'react';
+
+import Header from '../../../../common/components/header';
+import Table from '../../../../common/components/table';
+import {getPolicyById} from '../../../../common/services';
+import {
+	deleteClaimByExternalReferenceCode,
+	getClaims,
+} from '../../../../common/services/Claim';
+import formatDate from '../../../../common/utils/dateFormatter';
+
+const ClaimsTable = () => {
+	const [dataClaims, setDataClaims] = useState<TableContentType[]>([]);
+	const [totalPages, setTotalPages] = useState<number>(0);
+	const [pageSize, setPageSize] = useState<number>(20);
+	const [totalCount, setTotalCount] = useState<number>(0);
+	const [page, setPage] = useState<number>(1);
+	const [firstPaginationLabel, setFirstPaginationLabel] = useState<number>(1);
+	const [secondPaginationLabel, setSecondPaginationLabel] = useState<number>(
+		1
+	);
+
+	const HEADERS = [
+		{
+			greyColor: true,
+			key: 'dateCreated',
+			value: 'Date Field',
+		},
+		{
+			greyColor: true,
+			key: 'productName',
+			value: 'Product',
+		},
+		{
+			bold: true,
+			key: 'id',
+			type: 'link',
+			value: 'Claim Number',
+		},
+		{
+			greyColor: true,
+			key: 'policyNumber',
+			value: 'Policy Number',
+		},
+		{
+			greyColor: true,
+			key: 'claimName',
+			value: 'Name',
+		},
+		{
+			greyColor: true,
+			key: 'claimStatus',
+			type: 'status',
+			value: 'Status',
+		},
+	];
+
+	const PARAMETERS = {
+		page: '0',
+		pageSize: '0',
+	};
+
+	PARAMETERS.pageSize = pageSize.toString();
+	PARAMETERS.page = page.toString();
+
+	type TableContentType = {
+		[key: string]: string;
+	};
+
+	type ClaimTableType = {
+		claimStatus: {name: string};
+		dateCreated: string;
+		externalReferenceCode: string;
+		id: string;
+		r_policyToClaims_c_raylifePolicyId: any;
+	};
+
+	async function getProductName(policyId: number) {
+		const policy = await getPolicyById(policyId);
+
+		return policy?.data?.productName;
+	}
+
+	async function getPolicyOwnerName(policyId: number) {
+		const policy = await getPolicyById(policyId);
+
+		return policy?.data?.policyOwnerName;
+	}
```

Comment:
We have two very similar functions here. What do you think to turn it into a generic one?